docs/proposals/kuratorcd/kurator-cd.md

```diff
@@ -0,0 +1,622 @@
+    // regardless of any changes to its target, services, etc. Note that if the
+    // Canary is suspended during an analysis, its paused until the Canary is unsuspended.
+    // +optional
+    Suspend bool `json:"suspend,omitempty"`
+}
+```
+
+TargetObjectReference contains enough information to let you locate the typed referenced object in the same namespace. The two types of Kind now supported are `Deployment` and `DaemonSet`.
+
+```concole
+// Note: refer to https://github.com/fluxcd/flagger/blob/main/pkg/apis/flagger/v1beta1/canary.go
+type TargetObjectReference struct {
+    // API version of the referent
+    // +optional
+    APIVersion string `json:"apiVersion,omitempty"`
+
+    // Kind of the referent
+    // +optional
+    Kind string `json:"kind,omitempty"`
+
+    // Name of the referent
+    Name string `json:"name"`
+}
+```
+
+The ServiceConfig is the configuration of the service that is generated according to the object specified in the TargetObject. The main thing is to specify the `Port` and `targetPort`.
+
+```console
+// Note: refer to https://github.com/fluxcd/flagger/blob/main/pkg/apis/flagger/v1beta1/canary.go
+type ServiceConfig struct {
+    // Name of the virtual Kubernetes generated by Flagger.
+    // Defaults to DeliveryPolicy.TargetObject.Name
+    // +optional
+    Name string `json:"name,omitempty"`
+
+    // Port of the generated Kubernetes service.
+    Port int32 `json:"port"`
+
+    // Port name of the generated Kubernetes service.
+    // Defaults to http
+    // +optional
+    PortName string `json:"portName,omitempty"`
+
+    // Target port number or name of the generated Kubernetes service.
+    // Defaults to CanaryService.Port
+    // +optional
+    TargetPort intstr.IntOrString `json:"targetPort,omitempty"`
+
+    // AppProtocol of the service.
+    // https://kubernetes.io/docs/concepts/services-networking/service/#application-protocol
+    // +optional
+    AppProtocol string `json:"appProtocol,omitempty"`
+
+    // PortDiscovery adds all container ports to the generated Kubernetes service.
+    // Defaults to true
+    PortDiscovery bool `json:"portDiscovery"`
+}
+```
+
+Kurator will create a VirtualService resource based on the configuration in `VirtualServiceConfig` to distribute traffic.
+
+```console
+// Note: refer to https://github.com/fluxcd/flagger/blob/main/pkg/apis/flagger/v1beta1/canary.go
+type VirtualServiceConfig struct {
+    // Timeout of the HTTP or gRPC request.
+    // +optional
+    Timeout string `json:"timeout,omitempty"`
+
+    // Gateways attached to the generated Istio virtual service.
+    // Defaults to the internal mesh gateway.
+    // +optional
+    Gateways []string `json:"gateways,omitempty"`
+
+    // Hosts attached to the generated Istio virtual service.
+    // Defaults to the service name
+    // +optional
+    Hosts []string `json:"hosts,omitempty"`
+
+    // If enabled, Flagger would generate Istio VirtualServices without hosts and gateway,
```

Comment:
nit, Annotations should start with the parameter name, followed by its description. For example, 'Delegation [description of Delegation]'